client/public/index.html

```diff
@@ -28,6 +28,7 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
     <title>React App</title>
+    <script type="text/javascript" src="https://maps.googleapis.com/maps/api/js?key={API_KEY}&libraries=places"></script>
```

Comment:
Where does `API_KEY` get set?

Reply:
It doesn't. Right now it's up to the dev to change it locally/on the VM. I just wanted to get rid of your (now invalidated) API key from the repo.

Reply:
Ok. I think we should set it up so `.env` files can be used to fill this in automatically. Take a look at this thread https://stackoverflow.com/questions/51744375/create-react-app-using-environment-variables-in-index-html

Reply:
I think we will basically need something like this in replacement of the script tag, in order to use an environment variable:

```html
<div id="#googleMapsScriptPlaceholder"/>
<script type="text/javascript">
    var el = document.createElement('script');
    el.src = "https://maps.googleapis.com/maps/api/js?key=%GOOGLE_API_KEY%&libraries=places";
    document.getElementById("#googleMapsScriptPlaceholder").replaceWith(el);
</script>
```

Then add `.env` to `gitignore` and create the `.env` file locally
```
GOOGLE_API_KEY=blah
```

Reply:
https://create-react-app.dev/docs/adding-custom-environment-variables/

> WARNING: Do not store any secrets (such as private API keys) in your React app!

> Environment variables are embedded into the build, meaning anyone can view them by inspecting your app's files.

Reply:
From all of the threads I've read, half of the people say "DON'T DO THIS IT'S NOT SECURE" and the other half ignore that it's a potential problem, lol. Many suggest making a request to the back-end to the get the API key.

Thoughts of doing that in conjunction with your 2nd example above where we insert a placeholder `<div>` then replace it using JS?

Reply:
Duh how did I miss that. I've used .env in node before so it was secure but it makes sense that it's not here.

I still think it makes sense to use it here because we can have two keys then: one that is unrestricted and used locally, which cannot be exposed, and one that is restricted to the monumental.se.rit.edu domain and can be exposed and is only used on the VM since it shouldn't work locally hopefully

Reply:
You're able to restrict the API key's usage by domain through the Google console?

Reply:
![image](https://user-images.githubusercontent.com/25756457/68105881-deaa2280-fead-11e9-823d-62d9ebbba7dd.png)

Reply:
Fancy.

Reply:
Alright, I've added a `.env` file to `client/` that we can use to set the API key. Right now, it's just set to `VmApiKey`. Once this is deployed to the VM, we'll change this to the restricted VM only key and add this file to the `.gitignore` so we can change it locally to be the unrestricted key.

Reply:
I don't think the file should be included in the repo. We should just have a private gist with the VM's values stored.